src/screens/CreatePost/index.js

```diff
@@ -449,11 +450,13 @@ const CreatePost = () => {
   };
 
   const postV2 = async () => {
+    setLoadingPost(true);
     if (!isEmptyMessageAllowed()) {
       showMessage({
         message: StringConstant.createPostFailedNoMessage,
         type: 'danger'
       });
+      setLoadingPost(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `setLoadingPost(true)` before checking if the message is empty and then setting it back to false if the message is indeed empty is a bit inefficient. You could check if the message is empty first and only set the loading state if the message is not empty.
```diff
-    setLoadingPost(true);
     if (!isEmptyMessageAllowed()) {
       showMessage({
         message: StringConstant.createPostFailedNoMessage,
         type: 'danger'
       });
-      setLoadingPost(false);
       return true;
     }
+    setLoadingPost(true);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->